### PR TITLE
[finishes #187354262] Buyer can provide review on sold product when order is delivered

### PR DESCRIPTION
**Pull request by amin-leon**
## Purpose

This task aims to give a space for buyer to provide a review(rating + feedback) for product he/she has already bought and received. 

## Changes Made

- Created a review model.
- Created a route to provide a review on product purchased and delivered.
- Created a route to view all reviews for a specific product.
- Created a route to delete reviews on a specific product only done by Owner (buyer) of the review.
- Created a route to view all reviews statistics(average ratings) for a specific product.
- swagger api documentation

## Testing Instructions

Follow provided routes by providing all parameters as required

## Related Issues

None

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:

- [x] Code follows the project's coding standards
- [x] Changes are covered by tests
- [x] Documentation is updated (if applicable)
- [x] All tests pass

### CODE REVIEW

File: src/controllers/productsController.ts [hunk 1]
```diff
@@ -13,6 +13,9 @@ import User from '../database/models/user';
 import { sendEmail } from '../helpers/send-email';
 import { destroyImage } from '../helpers/destroyImage';
 import { extractImageId } from '../helpers/extractImageId';
+import Review, { ReviewAttributes } from '../database/models/Review';
```
Comment (niyontwali) [inline src/controllers/productsController.ts:17]: Could you please rename the file from "Review" to "review"? I know some others did not follow the convention, but that will be resolved in the bug fix features.

File: src/controllers/productsController.ts [hunk 2]
```diff
@@ -355,3 +358,159 @@ Product.afterCreate(async product => {
...
+export const deleteReview = async (req: Request, res: Response) => {
+  try {
+    const { reviewId } = req.params;
+    const buyer = req.user as any;
+    const userId = buyer.id;
+
+    // Check if the review exists
+    const review = await Review.findByPk(reviewId);
+    if (!review) {
+      return res.status(404).json({ message: 'Review not found' });
```
Comment (niyontwali) [inline src/controllers/productsController.ts]: Please remember to do our json object like this 

```
{
   ok: false,
   message: 'Review not found'
}
```

File: src/controllers/productsController.ts [hunk 3]
```diff
@@ -355,3 +358,159 @@ Product.afterCreate(async product => {
...
+    }
+
+    // verify owner
+    if (review.userId !== userId) {
+      return res.status(400).json({ message: 'Only owner can delete a review' });
```
Comment (niyontwali) [inline src/controllers/productsController.ts]: The same should be aplied here, the json should have the **ok** property as we agreed on it as a whole team

File: src/controllers/productsController.ts [hunk 4]
```diff
@@ -355,3 +358,159 @@ Product.afterCreate(async product => {
...
+
+    res.status(201).json({
+      ok: true,
+      message: 'Thank you for providing feedback!',
+    });
+  } catch (error) {
+    logger.error('Unable to provide review:', error);
+    sendInternalErrorResponse(res, error);
+  }
+};
```
Comment (niyontwali) [inline src/controllers/productsController.ts]: I believe the cartId is not necessary. Could you please clarify why we need it? Our implementation should handle the cart such that after a product is purchased (order made), the cart is either emptied (if all items are purchased) or the purchased item is removed from the cart. As long as we have the productId and its status based on the user's order, the user should be able to provide a review.

Typically, in most e-commerce systems, when an order is placed, the system stores the order details, including the user ID, product IDs, and order status (e.g., delivered). When a user wants to review a product, the system checks if the user has an order with a "delivered" status that includes the product being reviewed and verifies that it is the user's order. Once these conditions are met, the user should be able to provide a review.

**Review by niyontwali — CHANGES_REQUESTED** [review]
Please address the requested changes, and then we shall merge it. You might need to wait a little bit for @P-Rwirangira to have his task of tracking the status of the order merged

**Review by JeanIrad — APPROVED** [review]

**Review by patrickhag — APPROVED** [review]
LGTM

**Review by P-Rwirangira — APPROVED** [review]

**Review by hozayves — APPROVED** [review]

**Review by niyontwali — APPROVED** [review]

**Pull request merged**